### PR TITLE
Use OutputTypeEx property to get the Output Type in Adv Build Prop page

### LINKED ISSUE #1305: "library base address" is always disabled for library projects
1. File -> New Class Library (.NET Standard)
2. Choose Build -> Advanced

Expected: To be able to change the base address
Actual:

image

### PR DESCRIPTION
**Pull request by basoundr**
Fixes #1305

@dotnet/project-system for review

### CODE REVIEW

File: src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb [hunk 1]
```diff
@@ -192,12 +192,10 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
 
         Private Function IsExeProject() As Boolean
 
-            Dim obj As Object = Nothing
             Dim OutputType As VSLangProj.prjOutputType
 
             Try
-                GetCurrentProperty(VsProjPropId.VBPROJPROPID_OutputType, "OutputType", obj)
-                OutputType = CType(obj, VSLangProj.prjOutputType)
+                OutputType = OutputTypeProperty
```
Comment (srivatsn) [inline src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb]: How does this affect desktop projects?

Comment (basoundr) [inline src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb]: No, it does not. The reason is, `OutputTypeProperty` tries to get the outputtype from `ProjectProperties`, supported by Desktop projects, first before going through CPS.

Comment (davkean) [inline src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb]: Why doesn't this work already? It's a sign that we've broken compat.

Comment (RaulPerez1) [inline src/Microsoft.VisualStudio.Editors/PropPages/AdvBuildSettingsPropPage.vb]: I made a comment about this in one of the other bugs after investigating a similar issue. We're dying a death of a thousand cuts here and working around lil bits and pieces. I totally agree that we should implement ProjectProperties on the project system which would bring us back to full compat and eliminate a lot of these issues.

**Comment by davkean** [comment]
This isn't relevant given your new changes right @basoundr? I close it, reopen if it still is.

**Comment by basoundr** [comment]
Yup, with the new changes, it is not required.

**Pull request closed without merging**